### ISSUE TITLE
[Bug] x and z are swapped in Chunk::block

### ISSUE THREAD

**Issue by Badel2** [owengage/fastnbt#18]
I found this when trying to implement #17. When I use Chunk::block(x, y, z) to find certain blocks (iterate over all the blocks and print the coordinates of "minecraft:spawner"), it returns the wrong coordinates. For example, the x coordinate is off by +2 and the z coordinate is off by -2. Swapping x and z (Chunk::block(z, y, x) results in the correct coordinates.

I'm fairly sure that this is not a bug in my code, but I would advise to try to implement #17 first to confirm that this bug exists. Probably all the functions that use in-chunk block coordinates (values from 0 to 15) are wrong.

**Comment by owengage** [comment]
Hi, thanks for pointing this out, I will take a look when I get chance! :)

**Comment by owengage** [comment]
This turns out to be a problem all the way from the start of the project. Basically a big in the `anvil` executable plus the desire to make a map look right.

Here's the real cause:

```diff
--- a/tools/src/bin/anvil.rs
+++ b/tools/src/bin/anvil.rs
@@ -40,8 +40,8 @@ fn region_paths(in_path: &Path) -> Result<Vec<PathBuf>> {
 fn coords_from_region(region: &Path) -> Option<(isize, isize)> {
     let filename = region.file_name()?.to_str()?;
     let mut parts = filename.split('.').rev().skip(1);
-    let z = parts.next()?.parse::<isize>().ok()?;
     let x = parts.next()?.parse::<isize>().ok()?;
+    let z = parts.next()?.parse::<isize>().ok()?;
     Some((x, z))
 }
```

The x and z were flipped when parsing the region file names. This led me to assume the order of everything else was inverted. I've gone through an fixed them all hopefully. Will push very soon.

**Comment by owengage** [comment]
Didn't intend to close. Fix is on master but don't have the time to release right now. Will do it tomorrow/tonight.

Thanks for discovering my entire coordinate system was inverted!

**Comment by Badel2** [comment]
Thanks for the quick reply! I will confirm if the fix works as indented later today.

**Comment by owengage** [comment]
Published 0.11 of both fastanvil and fastnbt, which should fix it. If for some reason it doesn't don't hesitate to open another issue.

This literally flipped my world upside down (and rotated it by 90 degrees) :)

**Comment by Badel2** [comment]
0.11 working great, thanks again!
